src/Symfony/Component/Messenger/Transport/AmqpExt/AmqpSender.php

```diff
@@ -41,7 +42,11 @@ public function send(Envelope $envelope): Envelope
     {
         $encodedMessage = $this->serializer->encode($envelope);
 
-        $this->connection->publish($encodedMessage['body'], $encodedMessage['headers']);
+        try {
+            $this->connection->publish($encodedMessage['body'], $encodedMessage['headers']);
+        } catch (\AMQPException $e) {
+            throw new TransportException('Current transport was not able to send given message, please try again', 0, $e);
```

Comment:
maybe use `$e->getMessage()` if the base message is clear enough?

Reply:
The base message is pretty clear yes, but it's coupled with amqp. Wouldn't be better to access it from $previous if needed and have a default component's message?

Reply:
Putting the most relevant message on top of the stack helps debug faster. I would use the original message.

Reply:
I agree with Nicolas, let's use `$e->getMessage()`